### DISCUSSION TITLE
es-module-lexer + cjs-module-lexer + oxc parser

### DISCUSSION THREAD

**Discussion by Boshen** [oxc-project/oxc#2608] [Ideas]
es-module-lexer and cjs-module-lexer doesn't handle TypeScript files, we can offer a faster tool.

https://github.com/oxc-project/oxc/issues/623

**Comment by Boshen** [comment]
* https://crates.io/crates/oxc_module_lexer
* https://www.npmjs.com/package/oxc-parser (exports module_lexer)
* cjs-module-lexer is too complicated.

**Comment by SukkaW** [comment]
@Boshen Hi, I'd like to know if it is possible to access lexer through `oxc-parser` now?

I am looking for a lexer that works across all CJS/ESM/JSX/TS/TSX to scan imports/exports.

I am experimenting with a few ideas to improve performance for those who are still sticking with ESLint: 
https://github.com/un-ts/eslint-plugin-import-x/issues/201

**Reply by Boshen** [reply]
Only ESM works right now, TS in the future. CJS will not be planned.

**Reply by SukkaW** [reply]
> Only ESM works right now, TS in the future. CJS will not be planned.

Thanks for your reply! I can still use `cjs-module-lexer` for CJS, but IMHO it would be nice to have one lexer for all.
